src/github.com/couchbase/sync_gateway/db/index_changes.go

```diff
@@ -168,13 +117,19 @@ func (db *Database) VectorMultiChangesFeed(chans base.Set, options ChangesOption
 				if minEntry.Seq.TriggeredBy == 0 {
 					// Update the cumulative clock, and stick it on the entry.
 					cumulativeClock.SetMaxSequence(minEntry.Seq.vbNo, minEntry.Seq.Seq)
-					// TODO: Potential enhancement - for first iteration loop, always set hash on last entry.
-					continuousLastHash = db.calculateHashWhenNeeded(
+					// Force new hash generation for non-continuous changes feeds if this is the last entry to be sent - either
+					// because there are no more entries in the channel feeds, or we're going to hit the limit.
+					forceHash := false
+					if options.Continuous == false && (nextEntry == nil || options.Limit == 1) {
+						forceHash = true
```

Comment:
In general the design of VectorMultiChangesFeed (and the original MultiChangesFeed in changes.go that it's based on) expects to mutate the incoming options - in particular it's also done with options.Since.  

Having said that, this does make the processing more difficult to follow, and I think we should refactor in future.  In the last few commits to index_changes I think we've been making some decent progress in moving away from the huge monolithic changes loop code block - in 1.3 I'm planning to continue that, and also replicate that refactoring in the original changes.go.  

I'll file a ticket to track this work in 1.3.